plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx

```diff
@@ -156,6 +164,19 @@ const ProductTemplateEdit = ( {
 		12,
 		isNumber
 	);
+
+	// Add default query context attributes to queryContextIncludes
+	queryContextIncludes = [
+		...new Set(
+			queryContextIncludes.concat( DEFAULT_QUERY_CONTEXT_ATTRIBUTES )
+		),
+	];
+
+	const productCollectionQueryContext = useProductCollectionQueryContext( {
+		clientId,
+		queryContextIncludes,
+	} );
+
```

Comment:
I suggest not to modify the props value directly, especially since the `queryContextIncludes` may be `undefined` for existing Product Collection blocks:

<img width="1521" alt="image" src="https://github.com/woocommerce/woocommerce/assets/20098064/5b8b8bcf-01ff-4061-9469-cc61f552a20e">


```suggestion
	// Add default query context attributes to queryContextIncludes
	const allQueryContextIncludes = queryContextIncludes ? [
		...new Set(
			queryContextIncludes.concat( DEFAULT_QUERY_CONTEXT_ATTRIBUTES )
		),
	] : [ DEFAULT_QUERY_CONTEXT_ATTRIBUTES ];

	const productCollectionQueryContext = useProductCollectionQueryContext( {
		clientId,
		allQueryContextIncludes,
	} );

```

Reply:
Good catch! I have made the changes in https://github.com/woocommerce/woocommerce/pull/44150/commits/30aec35a05bd83c358681450dc88a64b8807fb3c. Your attention to detail during the PR review is truly appreciated 🙌🏻